src/Drupal/DKANExtension/Context/WorkflowContext.php

```diff
@@ -30,23 +35,27 @@ public function transitionModerationState($named_entity, $state, $date = null) {
       throw new \Exception("State '$state' is not available. All possible states are [$possible_states].");
     }
 
-    $current_user = $this->getCurrentUser();
+    $current_user = $user = ($workflow_user) ? user_load_by_name($workflow_user) : $this->getCurrentUser();
     if (!$current_user) {
       throw new \Exception("No user is logged in.");
     }
 
-    $my_revision = $node->workbench_moderation['my_revision'];
-    $next_states = workbench_moderation_states_next($my_revision->state, $current_user, $node);
-    if (empty($next_states)) {
-      $next_states = array();
-    }
-    if (!isset($next_states[$state_key])) {
-      $next_states = implode(", ", $next_states);
-      throw new \Exception("State '$possible_states[$state_key]' is not available to transition to. Transitions available to user '$current_user->name' are [$next_states]");
-    }
+    $my_revision = $node->workbench_moderation['my_revision'];;
+    $flipped_posible_states = array_flip($possible_states);
```

Comment:
The array has this shape:

array( 'needs_review' => 'Needs Review', ...).

Since I need the machine name (later in the next comment) but I have the label then I flip it. In other words I don't need to now if the array contains the element but get the mapped machine name from the label.